yew/src/virtual_dom/vtag.rs

```diff
@@ -110,6 +118,63 @@ impl Clone for VTag {
     }
 }
 
+#[cfg(feature = "ssr")]
+impl ToHtmlString for VTag {
+    fn to_html_string(&self) -> String {
+        let mut parts: Vec<String> = Vec::new();
+        let tag_name = htmlescape::encode_minimal(&self.tag).to_lowercase();
+        parts.push(format!("<{}", tag_name).to_string());
+
+        for (key_unclean, value) in &self.attributes {
+            let key = key_unclean.to_lowercase();
+            // checked, value (special if textarea), disabled, href?, selected,
+            // kind -> type if input, disallow ref, disallow LISTENER_SET, class
+
+            let mut is_valid = true;
+            for c in key.chars() {
+                let is_alnum = (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9');
+                if !is_alnum && c != '-' && c != '_' && c != ':' {
+                    is_valid = false;
+                    warn!("Invalid attribute name: `{}`", &key);
+                    break;
+                }
+            }
+
+            let is_skipped = tag_name == "textarea" && key == "value";
+
+            if is_valid && !is_skipped {
+                parts.push(
+                    format!(" {}=\"{}\"", key, htmlescape::encode_attribute(&value)).to_string(),
+                );
+            }
+        }
+
+        if self.checked {
+            parts.push(" checked=\"true\"".to_string())
+        }
+
+        if tag_name == "input" {
+            if let Some(kind) = &self.kind {
+                parts
+                    .push(format!(" type=\"{}\"", htmlescape::encode_attribute(&kind)).to_string());
```

Comment:
It would be nice if htmlescape allowed whitelisting of certain characters, so we can get a space instead of `&#x20;` in e.g. class attributes. Might need to make an upstream PR for this, although `htmlescape` seems as though it may be abandonware.

Reply:
I think it actually does not belong in `htmlescape` after all. The reason they escape so many innocuous characters is because of OWASP recommendations to do so because of developers often not quoting the attribute values. As such, I think it should be more on the consumers of such a library to do any whitelisting. That said, it's not really necessary for this use case, since any SSR-generated code isn't necessarily meant to be read anyway.

Reply:
> although htmlescape seems as though it may be abandonware.

I've looked at it (very briefly) and it doesn't seem to be a very big project – could probably maintain a fork/adopt it.